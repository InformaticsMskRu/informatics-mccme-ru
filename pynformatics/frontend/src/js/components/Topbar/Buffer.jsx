import React from 'react';
import styled from 'styled-components';

import Popover from '../utility/Popover';


const BufferWrapper = styled.div`
  cursor: pointer;
  display: flex;
`;


export default class Buffer extends React.Component {
  render() {
    return (
      <div>
        <Popover
          trigger="click"
          content={<div>buffer</div>}
        >
          <BufferWrapper>
            <i className="material-icons">work</i>
          </BufferWrapper>
        </Popover></div>
    );
  }
}
