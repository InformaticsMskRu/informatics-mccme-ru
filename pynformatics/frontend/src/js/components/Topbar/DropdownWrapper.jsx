import styled from 'styled-components';
import IsoDropdownWrapper from '../../isomorphic/components/topbar/topbarDropdown.style';


const DropdownWrapper = styled(IsoDropdownWrapper)`
  margin: -12px -16px;
`;

export default DropdownWrapper;
