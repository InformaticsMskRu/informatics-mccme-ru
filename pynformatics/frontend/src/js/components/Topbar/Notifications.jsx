import React from 'react';
import styled from 'styled-components';

import Popover from '../utility/Popover';


const NotificationsWrapper = styled.div`
  cursor: pointer;
  display: flex;
`;


export default class Notification extends React.Component {
  render() {
    return (
      <div>
      <Popover
        trigger="click"
        content={<div>notifications</div>}
      >
        <NotificationsWrapper>
          <i className="material-icons">notifications_active</i>
        </NotificationsWrapper>
      </Popover></div>
    );
  }
}
