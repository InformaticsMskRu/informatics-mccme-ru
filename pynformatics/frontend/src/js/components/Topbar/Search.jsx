import React from 'react';
import styled from 'styled-components';
import { Icon } from 'antd';

import Input from '../utility/Input';


const SearchWrapper = styled.div`
  color: #ffffff;
  
  &&& .ant-input {
    background: rgba(129, 132, 145, 0.3);
    border: none;
    border-radius: 15px;
    color: #ffffff;
    padding: 0 33px 0 37px;
    min-width: 284px;
    height: 30px;
    font-family: Roboto, sans-serif;
    font-size: 15px;
    
    ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
      color: rgba(255, 255, 255, 0.3);
    }
    ::-moz-placeholder { /* Firefox 19+ */
      color: rgba(255, 255, 255, 0.3);
    }
    :-ms-input-placeholder { /* IE 10+ */
      color: rgba(255, 255, 255, 0.3);
    }
    :-moz-placeholder { /* Firefox 18- */
      color: rgba(255, 255, 255, 0.3);
    }
  }
  
  .ant-input-prefix i {
    color: rgba(255, 255, 255, 0.3);
    cursor: default;
  }
  .ant-input-suffix i {
    color: #ffffff;
    cursor: pointer;
  }
`;

export default class Search extends React.Component {
  constructor() {
    super();
    this.state = {
      searchQuery: '',
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({
      searchQuery: e.target.value,
    })
  }

  render() {
    const { searchQuery } = this.state;

    const prefix = <i className="material-icons md-18">search</i>;
    const suffix = searchQuery ? <Icon type="arrow-right" /> : '';
    const placeholder = 'Найти задачу, курс или сборы';

    return (
      <SearchWrapper>
        <Input
          onChange={this.handleChange}
          prefix={prefix}
          suffix={suffix}
          placeholder={placeholder}
        />
      </SearchWrapper>
    );
  }
};
