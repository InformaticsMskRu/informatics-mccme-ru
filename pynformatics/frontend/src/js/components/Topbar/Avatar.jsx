import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import Popover from '../utility/Popover';
import DropdownWrapper from './DropdownWrapper';
import isUserLoggedIn from '../../utils/isUserLoggedIn';


const AvatarWrapper = styled.div`
  cursor: pointer;
  display: flex;
  
  img {
    width: 30px;
    height: 30px;
    border-radius: 15px;
    background: #FFFFFF;
  }
`;


export class Avatar extends React.Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
  };

  constructor() {
    super();

    this.state = {
      popoverVisible: false,
    };

    this.togglePopoverVisible = this.togglePopoverVisible.bind(this);
  }

  togglePopoverVisible() {
    this.setState({...this.state, popoverVisible: !this.state.popoverVisible})
  }

  render() {
    const loggedIn = isUserLoggedIn(this.props.user);

    const content = loggedIn
      ? (
        <DropdownWrapper className="isoUserDropdown">
          <Link 
            className="isoDropdownLink"
            to="/" 
          >
            Профиль
          </Link>
          <Link 
            className="isoDropdownLink"
            to="/auth/logout"
            onClick={this.togglePopoverVisible}
          >
            Выйти
          </Link>
        </DropdownWrapper>
      ) : (
        <DropdownWrapper className="isoUserDropdown">
          <Link 
            className="isoDropdownLink"
            to={{
              pathname: '/auth/login',
              state: { from: this.props.location.pathname },
            }}
            onClick={this.togglePopoverVisible}
          >
            Войти
          </Link>
        </DropdownWrapper>
      );

    return (
      <Popover
        trigger="click"
        content={content}
        placement="bottomLeft"
        arrowPointAtCenter={true}
        visible={this.state.popoverVisible}
        onVisibleChange={this.togglePopoverVisible}
      >
        <AvatarWrapper>
          { loggedIn
            ? <img src="/images/ladybug.png"/>
            : <img src="/images/dog.png"/> }
        </AvatarWrapper>
      </Popover>
    );
  }
}


export default withRouter(connect(state => ({
  user: state.user,
}))(Avatar));
