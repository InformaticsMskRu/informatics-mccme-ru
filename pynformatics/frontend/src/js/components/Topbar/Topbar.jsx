import React from 'react';
import styled from 'styled-components';
import { Layout } from 'antd';
import { connect } from 'react-redux';

import IsoTopbarWrapper from '../../isomorphic/containers/Topbar/topbar.style';

import Avatar from './Avatar';
import Buffer from './Buffer';
import Button from '../utility/Button';
import Notifications from './Notifications';
import Search from './Search';
import * as uiActions from '../../actions/uiActions';

const { Header } = Layout;

const TopbarWrapper = styled(IsoTopbarWrapper)`
  .isomorphicTopbar {
    &.collapsed {
      padding: 0 61px 0 68px;
    }
    &:not(.collapsed) {
      padding: 0 61px 0 68px;
      @media (min-width: 1280px) {
        padding-left: calc(68px + 320px);
      }
      @media (max-width: 767px) {
        padding: 0 15px !important;
      }
    }
    @media (max-width: 767px) {
      padding: 0 15px;
    }
    
    .isoLeft {
      flex: 2;
      
      .triggerBtn {
        &:before {
          font-size: 36px;
        }
      }
    }
    
    .isoRight {
      flex: 2;
      justify-content: flex-end;
      
      & > * {
        margin-left: 15px;
        margin-right: 15px;
      }
    }
    
    .leftTitle {
      font-size: 18px;
      margin-left: 19px;
    }
    .centerTitle {
      font-size: 18px;
    }
    
    .backBtn {
      margin-left: 28px;
    }
    
    .loginBtn > * {
      font-size: 14px;
      height: 34px; 
    }
    
    @media (max-width: 575px) {
      .bufferBtn, .notificationBtn { display: none; } 
      & > * { margin: 0; }
      .backBtn { margin-left: 28px; }
    }
    @media (max-width: 1279px) {
      .searchInput, .leftTitle { display: none; }
    }
    @media (min-width: 1280px) {
      .centerTitle, .backBtn { display: none; }
    }
  }
`;

export class Topbar extends React.Component {
  render() {
    const styling = {
      background: '#333b50',
      height: 60,
      width: '100%',
      position: 'fixed',
      color: '#ffffff',
    };

    const loggedIn = true;

    const rightContent = loggedIn ? [
      <div className="bufferBtn" key={0}><Buffer /></div>,
      <div className="notificationBtn" key={1}><Notifications /></div>,
      <div className="userAvatar" key={2}><Avatar /></div>,
    ] : [
      <div className="loginBtn"><Button type="primary" size={"small"}>ВОЙТИ</Button></div>
    ];


    return (
      <TopbarWrapper>
        <Header
          style={styling}
          className={
            this.props.sidebarCollapsed ? 'isomorphicTopbar collapsed' : 'isomorphicTopbar'
          }
        >

          <div className="isoLeft">
            <i
              className="material-icons md-24"
              onClick={() => this.props.dispatch(uiActions.toggleSidebar())}
              style={{ cursor: 'pointer' }}
            >
              menu
            </i>
            <i className="material-icons md-24 backBtn">keyboard_arrow_left</i>
            <div className="leftTitle">Информатикс</div>
          </div>

          <div className="centerTitle">Информатикс</div>

          <div className="isoRight">
            <div className="searchInput">
              <Search />
            </div>
            {rightContent}
          </div>
        </Header>
      </TopbarWrapper>
    );
  }
}

export default connect(state => ({
  sidebarCollapsed: state.ui.sidebarCollapsed,
}))(Topbar);
