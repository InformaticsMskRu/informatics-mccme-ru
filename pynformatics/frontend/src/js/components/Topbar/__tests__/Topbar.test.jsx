import React from 'react';
import renderer from 'react-test-renderer';

// Этот импорт вызывает ошибку
// undefined:3:4584: missing '}'
// Которая ведет в IsoTopbarWrapper
// TODO: найти способ исправить
// import 'jest-styled-components';

import { Topbar } from '../Topbar';


describe('renders correctly', () => {
  it('with sidebar collapsed', () => {
    const tree = renderer.create(
      <Topbar sidebarCollapsed={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('with sidebar not collapsed', () => {
    const tree = renderer.create(
      <Topbar sidebarCollapsed={false} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
