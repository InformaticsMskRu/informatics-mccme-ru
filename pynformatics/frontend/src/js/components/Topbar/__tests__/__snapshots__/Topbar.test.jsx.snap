// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with sidebar collapsed 1`] = `
<div
  className="sc-jTzLTM dwFtUk sc-bdVaJa EJzqM"
  data-rtl={null}
>
  <div
    className="isomorphicTopbar collapsed ant-layout-header"
    style={
      Object {
        "background": "#333b50",
        "color": "#ffffff",
        "height": 60,
        "position": "fixed",
        "width": "100%",
      }
    }
  >
    <div
      className="isoLeft"
    >
      <i
        className="material-icons md-24"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        menu
      </i>
      <i
        className="material-icons md-24 backBtn"
      >
        keyboard_arrow_left
      </i>
      <div
        className="leftTitle"
      >
        Информатикс
      </div>
    </div>
    <div
      className="centerTitle"
    >
      Информатикс
    </div>
    <div
      className="isoRight"
    >
      <div
        className="searchInput"
      >
        <div
          className="sc-VigVT krbTSa"
        >
          <span
            className="sc-dnqmqq bXrAHQ ant-input-affix-wrapper"
            style={undefined}
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="material-icons md-18"
              >
                search
              </i>
            </span>
            <input
              className="ant-input"
              data-rtl={null}
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Найти задачу, курс или сборы"
              style={null}
              type="text"
            />
          </span>
        </div>
      </div>
      <div
        className="bufferBtn"
      >
        <div>
          <div
            className="sc-ifAKCX evHZck"
            onBlur={undefined}
            onClick={[Function]}
            onContextMenu={undefined}
            onFocus={undefined}
            onMouseDown={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            onTouchStart={[Function]}
          >
            <i
              className="material-icons"
            >
              work
            </i>
          </div>
        </div>
      </div>
      <div
        className="notificationBtn"
      >
        <div>
          <div
            className="sc-htoDjs bXpoql"
            onBlur={undefined}
            onClick={[Function]}
            onContextMenu={undefined}
            onFocus={undefined}
            onMouseDown={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            onTouchStart={[Function]}
          >
            <i
              className="material-icons"
            >
              notifications_active
            </i>
          </div>
        </div>
      </div>
      <div
        className="userAvatar"
      >
        <div
          className="sc-bxivhb iDuMfz"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={[Function]}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onTouchStart={[Function]}
        >
          <img
            src="invalid url"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with sidebar not collapsed 1`] = `
<div
  className="sc-jTzLTM dwFtUk sc-bdVaJa EJzqM"
  data-rtl={null}
>
  <div
    className="isomorphicTopbar ant-layout-header"
    style={
      Object {
        "background": "#333b50",
        "color": "#ffffff",
        "height": 60,
        "position": "fixed",
        "width": "100%",
      }
    }
  >
    <div
      className="isoLeft"
    >
      <i
        className="material-icons md-24"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        menu
      </i>
      <i
        className="material-icons md-24 backBtn"
      >
        keyboard_arrow_left
      </i>
      <div
        className="leftTitle"
      >
        Информатикс
      </div>
    </div>
    <div
      className="centerTitle"
    >
      Информатикс
    </div>
    <div
      className="isoRight"
    >
      <div
        className="searchInput"
      >
        <div
          className="sc-VigVT krbTSa"
        >
          <span
            className="sc-dnqmqq bXrAHQ ant-input-affix-wrapper"
            style={undefined}
          >
            <span
              className="ant-input-prefix"
            >
              <i
                className="material-icons md-18"
              >
                search
              </i>
            </span>
            <input
              className="ant-input"
              data-rtl={null}
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Найти задачу, курс или сборы"
              style={null}
              type="text"
            />
          </span>
        </div>
      </div>
      <div
        className="bufferBtn"
      >
        <div>
          <div
            className="sc-ifAKCX evHZck"
            onBlur={undefined}
            onClick={[Function]}
            onContextMenu={undefined}
            onFocus={undefined}
            onMouseDown={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            onTouchStart={[Function]}
          >
            <i
              className="material-icons"
            >
              work
            </i>
          </div>
        </div>
      </div>
      <div
        className="notificationBtn"
      >
        <div>
          <div
            className="sc-htoDjs bXpoql"
            onBlur={undefined}
            onClick={[Function]}
            onContextMenu={undefined}
            onFocus={undefined}
            onMouseDown={[Function]}
            onMouseEnter={undefined}
            onMouseLeave={undefined}
            onTouchStart={[Function]}
          >
            <i
              className="material-icons"
            >
              notifications_active
            </i>
          </div>
        </div>
      </div>
      <div
        className="userAvatar"
      >
        <div
          className="sc-bxivhb iDuMfz"
          onBlur={undefined}
          onClick={[Function]}
          onContextMenu={undefined}
          onFocus={undefined}
          onMouseDown={[Function]}
          onMouseEnter={undefined}
          onMouseLeave={undefined}
          onTouchStart={[Function]}
        >
          <img
            src="invalid url"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
